### DISCUSSION TITLE
Volumes lost after system went down. Why?

### DISCUSSION THREAD

**Discussion by Stitch10925** [seaweedfs/seaweedfs#3181] [Q&A]
I have been testing the resilience of SeaweedFS since I would love to use it in my homelab. Currently, I have 1 master and 4 volume servers, since 4 servers seems to be the minimum required for erasure encoding, which is what I was testing.

I added data to SeaweedFS and then systematically killed off volume servers to see if files were still working. After killing one of the servers everything was still working, this was to be expected, since EC would recover the missing chunks. After killing a second and third volume, I noticed that smaller files were not working anymore and parts of movies worked, and other parts didn't. This also makes sense, since EC does not have enough information to rebuild the chunks.

To my surprise, when I restarted the volume servers (I just turned them off to simulate a power loss), the volume that was still online managed to replicate its data to another volume server, however, on the other two volume servers all volumes and data seem to be gone. 

How is this possible? The master and filers remained online, and no data was deleted. The system was merely turned off.

Volumes are started as follows:
`./weed volume -mserver 192.168.xxx.xxx:9333 -dataCenter FP -index=leveldb`

**Comment by chrislusf** [comment]
Please make a reproducible case, possibly with docker compose.

**Reply by Stitch10925** [reply]
I have no experience with docker, so I'm not really sure how to set up a reproducible case to be honest.

What am I supposed to set up? 4 volume servers and a master?

**Reply by Stitch10925** [reply]
So, I tried it again.

I updated seaweedFs to the latest version, I then went through my configs. I noticed that I started the master with defaultReplication of 001, which I removed.

Then I removed all the filerdb folders on the volume servers and the master to get a clean setup. Then uploaded files to the instance again. This is my master.toml:

`scripts = """
  lock
  ec.encode -fullPercent=95 -quietFor=1h
  ec.rebuild -force
  ec.balance -force
  volume.deleteEmpty -quietFor=24h -force
  volume.balance -force
  volume.fix.replication
  s3.clean.uploads -timeAgo=24h
  unlock
"""
sleep_minutes = 15`

This is my startup for the master;

`./weed master &
./weed filer -master 127.0.0.1:9333 &
./weed mount -filer 127.0.0.1:8888 -dir=/mnt/samba -filer.path=/`

This is how I start my volumes:

`./weed volume -mserver <masterip>:9333  -dataCenter FP -index=leveldb`

First thing I notice is that the Master GUI shows no ErasureCodingShards. When I stopped one of the 4 volume some files became unavailable. The master would also get stuck in a retry-loop to open the file.

Obviously I have done something wrong since I don't think the EC is set up correctly.

----
Update:

After I restarted the volume that I killed, I see that it contains no volumes nor volumeIds anymore. Also after stopping the master, filer and mount, the filer on port 8888 now does not show any files anymore. It seems like the filer data is not being persisted.
